multidid/src/lib.rs

```diff
@@ -0,0 +1,342 @@
+use integer_encoding::*;
+use std::str;
+use std::collections::HashMap;
+use multibase::Base;
+
+/**
+ * Multicodec Codes https://github.com/multiformats/multicodec/blob/master/table.csv
+ */
+ const MULTIDID_CODEC: u32 = 0xd1d; // did
+ const ANY_METHOD_CODE: u32 = 0x55; // raw
+ const PKH_METHOD_CODE: u32 = 0xca; // Chain Agnostic
+ const SECP256K1_CODE: u32 = 0xe7; // secp256k1-pub
+ const BLS12_381_G2_CODE: u32 = 0xeb; // bls12_381-g2-pub
+ const X25519_CODE: u32 = 0xec; // x25519-pub
+ const ED25519_CODE: u32 = 0xed; // ed25519-pub
+ const P256_CODE: u32 = 0x1200; // p256-pub
+ const P384_CODE: u32 = 0x1201; // p384-pub
+ const P521_CODE: u32 = 0x1202; // p521-pub
+ const RSA_CODE: u32 = 0x1205; // rsa-pub
+ const KEY_CODES: [u32; 8] = [SECP256K1_CODE, BLS12_381_G2_CODE, X25519_CODE, ED25519_CODE, P256_CODE, P384_CODE, P521_CODE, RSA_CODE];
+
+fn key_method_code_len(code: &u32, key_prefix: &[u8]) ->  Result<u16, &'static str> {
```

Comment:
I would look into the [anyhow](https://crates.io/crates/anyhow) crate that defines a very generic error type. It makes it easy to return errors from functions without having to use static strings as error messages or build your own type.

As an aside if you do need to build your own error type I typically use [thiserror](https://crates.io/crates/thiserror) to do so.